docs/concepts/ingress-classes.md

```diff
@@ -0,0 +1,181 @@
+an authentication plugin credential is _not_ translated directly: it is only
+translated into Kong configuration if a KongConsumer resource references it.
+
+Because they create Kong configuration indenpendent of any other resources,
+directly-translated resources require an ingress class, and their class must
+match the class configured for the controller. Referenced resources do not
+require a class, but must be referenced by a directly-translated resource
+matches the controller.
+
+### Adding class information to resources
+
+Most resources use a [kubernetes.io/ingress-class annoration][class-annotation]
+to indicate their class. There are several exceptions:
+
+- v1 Ingress resources have a dedicated `class` field.
+- Knative Services [use the class specifed][knative-class] by the
+  `ingress.class` key of the Knative installation's `config-network` ConfigMap.
+  You can optionally [override this on a per-Service basis][knative-override]
+  by adding a `networking.knative.dev/ingress.class` annotation to the Service.
+
+### Enabling support for classless resources
+
+Specifying a class is optional for some resources. Although specifying a class
+is recommended, you can instruct the controller to process resources without a
+class annotation using flags:
+
+- `--process-classless-ingress-v1beta1` instructs the controller to translate
+  v1beta1 Ingress resources with no class annotation.
+- `--process-classless-kong-consumer` instructs the controller to translate
+  KongConsumer resources with no class annotation.
+
+These flags are primarily intended for compatibility with older configuration
+(older versions of the Kong Ingress Controller had less strict class
+requirements, and it was common to omit class annotations). If you are creating
+new configuration and do not have older configuration without class
+annotations, recommended best practice is to add class information to Ingress
+and KongConsumer resources and not set the above flags. Doing so avoids
+accidentally creating duplicate configuration in other ingress controller
+instances.
+
+These flags do not _ignore_ `ingress.class` annotations: they allow resources
+with no such annotation, but will not allow resource that have a non-matching
+`ingress.class` annotation.
+
+## When to use a custom class
+
+Using the default `kong` class is fine for simpler deployments, where only one
+Kong Ingress Controller instance is running in a cluster. Changing the class is
+typical when:
+
+- You install multiple Kong environments in one cluster to handle different
+  types of ingress traffic, e.g. when using separate Kong instances to handle
+  traffic on internal and external load balancers, or deploying different types
+  of non-production environments in a single test cluster.
+- You install multiple controller instances that create configuration in
+  different Kong workspaces (using the `--kong-workspace` flag).
```

Comment:
These are separate, distinct deployment strategies, and while we do not cover them in detail here, they are not the same. In the first, there are multiple separate Kong clusters for different purposes, each managed by their own controller, with entirely separate bodies of configuration. In the second, there is a single Kong cluster managed by multiple controllers.

While I'll agree that that could warrant additional clarification, and have added it in e53e95a2a2bf5e2506eae657116ffdcaec742cf0, I'd like to bring some concerns over how this was raised: this wasn't phased as a question, a la "Wait, aren't both of these the same thing, both separate Kong clusters?", it was phrased as a statement, that these terms do not, in fact, mean different things. Approaching these points of confusion as a learning opportunity, rather than a chance to announce <Gordon Ramsay voice> "IT'S RAWWRONG" right out the door, will go a long way towards making me think these comments come from a sincere misunderstanding, rather than apparent extreme confidence that I really have no idea what I'm talking about.

I'll agree that some of these concepts are confusing: they reflect a wide array of deployment options because we do offer considerable flexibility to structure them to meet a complex range of customer needs and preferred architectures, and that was something I did indeed take into account when writing this section: the first line, which indicates users should use the default for simpler configurations is the content targeted at more inexperienced users, who don't know if they have a situation that warrants a custom class, and is meant to reassure them that they probably don't, and can leave the default in place.

The later sections, which go into more complex deployments, are not targeted at inexperienced users: this guide targets both inexperienced and more savvy users, so it does seek to strike a balance between both audiences. Users with those more complex deployments _are_ more experienced by virtue of using them: those deployments strategies aren't what we present in our basic examples elsewhere, and my expectation, based on practical experience working with novice users and guiding users that need those more complex solutions through them, is that users with those more complex deployments are aware they're using them, will recognize that the examples describe something they do or plan to do, and will take away that they need to override the default class.